File: scratch/test_api_query.py
import requests, json

BASE = "https://edusphere-erp.onrender.com/api/v1"

def test():
    # Login
    payload = {"email": "teacher1@edusphere.edu", "password": "edusphere"}
    r = requests.post(f"{BASE}/auth/login", json=payload)
    print("Login status:", r.status_code)
    data = r.json()
    if not data.get("success"):
        print("Login failed:", data)
        return
    
    token = data.get("token")
    headers = {"Authorization": f"Bearer {token}"}
    
    # Get classes
    r = requests.get(f"{BASE}/academic/classes", headers=headers)
    print("Classes status:", r.status_code)
    classes_data = r.json()
    print("Classes response:")
    print(json.dumps(classes_data, indent=2)[:2000])

    # Get students
    r = requests.get(f"{BASE}/students?limit=200", headers=headers)
    print("Students status:", r.status_code)
    students_data = r.json()
    print("Students response keys:", students_data.keys() if isinstance(students_data, dict) else "Not dict")
    if isinstance(students_data, dict) and "students" in students_data:
        students_list = students_data["students"]
        print(f"Total students fetched: {len(students_list)}")
        if students_list:
            print("First student sample:", json.dumps(students_list[0], indent=2))

if __name__ == "__main__":
    test()

File: scratch/test_api_query.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final base = 'https://edusphere-erp.onrender.com/api/v1';
  final email = 'priya.joshi@edusphere.edu';
  final pass = 'edusphere';

  print('Logging in as $email...');
  try {
    final loginRes = await http.post(
      Uri.parse('$base/auth/login'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'email': email,
        'password': pass,
      }),
    );

    if (loginRes.statusCode != 200) {
      print('Login failed: ${loginRes.statusCode} - ${loginRes.body}');
      return;
    }

    final loginData = jsonDecode(loginRes.body);
    final token = loginData['token'] ?? loginData['accessToken'] ?? loginData['data']?['token'];
    print('Login successful! Token: $token');

    final headers = {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer $token',
    };

    // Get students
    print('\nFetching students from API...');
    final studentsRes = await http.get(
      Uri.parse('$base/students?limit=200'),
      headers: headers,
    );

    if (studentsRes.statusCode != 200) {
      print('Failed to fetch students: ${studentsRes.statusCode} - ${studentsRes.body}');
      return;
    }

    final studentsData = jsonDecode(studentsRes.body);
    final students = studentsData['students'] ?? studentsData['data'] ?? studentsData;
    print('Total students fetched: ${students.length}');

    final Map<String, List<String>> classStudents = {};
    for (var s in students) {
      final user = s['user'] ?? s['User'] ?? {};
      final name = '${user['firstName'] ?? ''} ${user['lastName'] ?? ''}'.trim();
      final cls = s['currentClass'] ?? s['Class'] ?? {};
      final clsName = cls['name'] ?? 'No Class';
      final sec = s['section'] ?? s['Section'] ?? {};
      final secName = sec['name'] ?? 'No Section';

      classStudents.putIfAbsent('$clsName - $secName', () => []).add(name);
    }

    print('\n--- Students grouped by Class & Section in API ---');
    classStudents.forEach((key, list) {
      print('$key: ${list.length} students');
      for (var name in list.take(5)) {
        print('  - $name');
      }
    });

  } catch (e) {
    print('Error: $e');
  }
}
